src/creature_states_spdig.c

```diff
@@ -1726,7 +1773,6 @@ short creature_arms_trap(struct Thing *thing)
         ERRORLOG("Creature has invalid control structure!");
         return 0;
     }
-    struct Dungeon* dungeon = get_dungeon(thing->owner);
```

Comment:
Same here as before. For code consistency and readability, if functionality is all the same, we have the used structures placed above.